svelte-web-frontend/src/lib/ai/compute-motion-descriptors.ts

```diff
@@ -0,0 +1,190 @@
+import type { Pose2DPixelLandmarks } from  "$lib/webcam/mediapipe-utils";
+import { GetScaleIndicator, Get2DVector, QijiaMethodComparisonVectors, getMatricesMAE, getMatricesRMSE, getMagnitude2DVec } from "./EvaluationCommonUtils";
+
+/**
+ * Removes duplicate frame times from an array of user poses and corresponding frame times.
+ * @param {Pose2DPixelLandmarks[]} userPoses - Array of user poses.
+ * @param {number[]} frameTimes - Array of frame times corresponding to the poses.
+ * @returns {{ adjUserPoses: Pose2DPixelLandmarks[], uniqueFrameTimes: number[] }} An object containing adjusted user poses and unique frame times.
+ */
+export function removeDuplicateFrameTimes(userPoses: Pose2DPixelLandmarks[], frameTimes: number[]): { adjUserPoses: Pose2DPixelLandmarks[], uniqueFrameTimes: number[] } {
+    const uniqueFrameTimes = [];
+    const adjUserPoses = [];
+
+    for (let i = 0; i < frameTimes.length; i++) {
+        const currentTime = frameTimes[i];
+
+        // Check if the current frame time is not the same as the previous one
+        if (i === 0 || currentTime !== frameTimes[i - 1]) {
+            uniqueFrameTimes.push(currentTime);
+            adjUserPoses.push(userPoses[i]);
+        }
+    }
+
+    return { adjUserPoses, uniqueFrameTimes };
+}
+
+
+/**
+ * Calculates the standard deviation of the differences between adjusted user jerks and reference jerks.
```

Comment:
Ca you update this jsdoc comment to reflect the changes you've made to the function?

Reply:
Done. Good catch!